src/parser.cpp

```diff
@@ -77,68 +116,158 @@ namespace json_parser {
         return content;
     }
 
+    // Do not advance it_begin if it has reached the end 
+    void safeIteratorIncrement(std::string::const_iterator& it_begin, std::string::const_iterator& it_end) {
+        if (it_begin == it_end) {
+            return;
+        }
+        it_begin++;
+    }
+
     JsonObj parseIntoJson(std::string const& content) {
         auto it_begin = content.cbegin();
         auto it_end = content.cend();
         auto stack = std::stack<char>{};
-        JsonObj res = iterIntoJson(it_begin,it_end,stack);
+        JsonObj res = objectHandler(it_begin,it_end,stack);
         return res;
     }
 
-    // Insert an entry to the json object with given value string and attribute string 
+    // Convert the string into a JsonValue 
     // The value type recognition takes place here 
-    void insertEntry(std::string const& attribute, std::string const& value, json_parser::JsonObj& j_obj) {
+    JsonValue stringToValue(std::string const& value) {
         if (isdigit(value[0])) {
-            j_obj.emplace(attribute, std::move(parsing_trait<int>::intoJson(value)));
-        } else if (isalpha(value[0])) {
-            j_obj.emplace(attribute, std::move(parsing_trait<std::string>::intoJson(value)));
+            return std::move(parsing_trait<int>::intoJson(value));
+        // } else if (isalpha(value[0])) {
+        } else {
+            return std::move(parsing_trait<std::string>::intoJson(value));
         }
     }
 
-    // This updated version has improved performance by avoiding unnecessary copying 
-    JsonObj iterIntoJson(std::string::const_iterator& it_begin, std::string::const_iterator& it_end, std::stack<char>& stack) {
-        stack.push('{');
-        it_begin++;
-        std::string token{};
-        std::string attribute_buff{};
-        std::string value_buff{};
-        JsonObj res{};
+    // Currently no support for special characters such as '\n' and '[' 
+    // The string handler should pass the tailing '"' 
+    std::string stringHandler(std::string::const_iterator& it_begin, std::string::const_iterator& it_end, std::stack<char>& stack) {
+        Token token{};
+        while (it_begin != it_end) {
+            if (*it_begin == '"' && stack.top() == '"') { // End of string 
```

Comment:
switch can also be used here instead of if for a potential performance boost if you place more often ones on top, not 100% sure.